draft-ietf-acme-dns-account-01.mkd

```diff
@@ -62,47 +63,142 @@ systems or environments to handle challenge-solving for a single domain.
-`CNAME` record to this infrastructure. A domain name can only have one `CNAME`
-in DNS.
+This document specifies two new challenge types. `dns-02` and `dns-account-01`.
 
-This document specifies a new challenge type, `dns-account-01`. This challenge
-leverages the ACME Account Resource URL to present an account-unique stable
-challenge to an ACME server. This challenge allows any domain name to delegate
-its domain validation to more than one service through ACME account-unique DNS
-records.
-
-As now multiple labels can be used to prove domain control, and they depend on
-the ACME account, any number of them can be generated in advance, and then all
-required `CNAME` records can be created statically. The dynamic part of the
-label depends on the ACME account and not the account key, to allow for
-seamless account key rollover without the label changing. This ensures very
-long-lived labels, without any security considerations.
+This work follows all recommendations set forth in "Domain Control
+Validation using DNS" [I-D.draft-ietf-dnsop-domain-verification-techniques].
 
 This RFC does not intend to deprecate the `dns-01` challenge specified in
-{{!RFC8555}}. Since this new challenge does not modify or build on any
-pre-existing challenges, the ability to complete the `dns-account-01` challenge
-requires ACME server operators to deploy new changes to their codebase. This
-makes adopting and using this challenge an opt-in process.
+{{!RFC8555}}. Since these new challenges do not modify any pre-existing challenges,
+the ability to complete the `dns-02` or `dns-account-01` challenge requires ACME server
+operators to deploy new changes to their codebase. This makes adopting and using this
+challenge an opt-in process.
+
+## DNS-02
+
+The `dns-02` challenge adds onto `dns-01` by introducing a scoping mechanism
+to the domain validation label. This allows for the client to specify if the
+intended domain validation is for a specific host, a wildcard domain, or a domain
+and all of its subdomains.
+
+## DNS-ACCOUNT-01
+
+The `dns-account-01` challenge leverages the ACME Account Resource URL to present
+an account-unique stable challenge to an ACME server. This challenge allows any
+domain name to delegate its domain validation to more than one service through
+ACME account-unique DNS records.
+
+With this new challenge, domain validation of the same DNS name can be done through
+different validation labels. Since these validation labels will depend on the ACME
+account KID, any number of them can be generated in advance. This allows all
+required `CNAME` records for domain validation delegation to be constructed statically.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
+# DNS-02 Challenge
+
+When the identifier being validated is a domain name, the client can prove control of that domain by provisioning a `TXT` resource record containing a designated value for a specific authorization domain name.
```

Comment:
I'd avoid using "authorization domain name" here and throughout, just because that term has a very specific (and different) meaning in the Baseline Requirements. It wouldn't strictly be a conflict since those definitions obviously don't apply here, but I think avoiding the confusion would be good. Perhaps "challenge domain name"? Or just calling the prefix the "authorization domain label"?

Reply:
What do you think about `validation domain name`?

Reply:
Yep, I think that would be fine. The only reason I didn't suggest it is I assumed you had some other reason to move away from it :)